main.go

```diff
@@ -0,0 +1,93 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strings"
+	// "net/url"
+	"os"
+)
+
+func main() {
+	var page int
+	page = 1
+	// params := url.Values{}
+	// params.Add("page", page)
+	for i := 0; i <155; i++{
+		url := fmt.Sprint("https://www.balldontlie.io/api/v1/players?page=", page)
+		// resp, err := http.Get("https://www.balldontlie.io/api/v1/players"+params.Encode())
+		resp, err := http.Get(url)
+		if err != nil {
+		log.Fatalln(err)
+		}
+
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+		log.Fatalln(err)
+		}
+
+		// sb := string(body)
+		// fmt.Printf("%T", sb)
+		// fmt.Print(sb)
+
+
+		type Player struct {
+			Data [] struct {
+				ID int `json:"id`
+				First_name string `json:"first_name"`
+				Height interface{} `json:"height_feet"`
+				Height_inches interface{} `json:"heigh_inches"`
+				Last_name string `json:"last_name"`
+				Position string `json:"position"`
+				Team struct {
+					ID int `json:"id"`
+					Abbreviation string `json:"abbreviation"`
+					City string `json:"city"`
+					Conference string `json:"conference"`
+					Division string `json:"division"`
+					Full_name string `json:"full_name"`
+					Name string `json:"name"`
+				} `json:"team`
+				Weight interface{} `json:"weight_pounds"`
+			} `json:"data"`
+			Meta [] struct {
+				Pages int `json:"total_pages"`
+				Current_page int `json:"current_page"`
+				Next_page int `json:"next_page"`
+				Per_page int `json:"per_page"`
+				Total_count int `json:"total_count"`
+			} `json:"meta"`
+			}
+		
+		
+		var m Player
+		err = json.Unmarshal(body, &m)
+		// for i := 0; i < len(m.Data); i++ {
+		// 	fmt.Println(m.Data[i].Position)
+
+		// }
+		if len(os.Args) == 1 {
```

Comment:
Right idea, however it would be better to move this first so that we check if the arguments that have passed in are correct. If the person using this program hasn't passed in the correct argument, then we've wasted all 155 (soon to be 3) network calls that are going to happen.

To help separate this logic apart a bit, what I'd like you to do is create a new function called `getPlayers()` that returns a list of `Player` and also a possible `error`. This is the go way of saying "Hey something went wrong, here's an error so you can do something about it". e.g

```
func getPlayers() ([]Player, error) {
    url := fmt.Sprint("https://www.balldontlie.io/api/v1/players?page=", page)
    resp, err := http.Get(url)
    if err != nil {
        return nil, err
    }
    players := []Player{}
    // Do whatever other stuff you had to parse out the Response and get a list of Players
    return players, nil
}
```

Once you've done this, then you can call `getPlayers()` from your `main()` function, once you've made sure the user has passed through a valid argument, and then iterate over them to find the players you want to print out.